source/MaterialXRender/ShaderMaterial.h

```diff
@@ -34,6 +35,60 @@ class MX_RENDER_API ShadowState
     float ambientOcclusionGain = 0.0f;
 };
 
+/// @struct ShaderMaterialDefinition
```

Comment:
In this POC we add the concept of a ShaderMaterial definition and state, allowing multiple ShaderMaterial objects to share the same state (contain hwShader and glProgram)